### PR TITLE
support vc runtime static linkage

### PR DESCRIPTION
**Pull request by wenbingl**

### CODE REVIEW

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -271,6 +271,17 @@ jobs:
           ctest -C RelWithDebInfo --output-on-failure
         displayName: Run C++ native tests
 
+  - job: WindowsStaticVC
+    pool:
+      vmImage: 'windows-latest'
+
+    steps:
+      - script: |
+          call .\build.bat -DOCOS_ENABLE_CTEST=ON -DCMAKE_MSVC_RUNTIME_LIBRARY="MultiThreaded$<$<CONFIG:Debug>:Debug>"
```
Comment (wenbingl) [inline .pipelines/ci.yml:280]: the argument string seems not to be easily used, any better idea?

Comment (es0m) [reply .pipelines/ci.yml:280]: one could use a cmake option ... ;D

Comment (wenbingl) [reply .pipelines/ci.yml:280]: that could be a solution, but I noticed what protobuf did the same thing actually brought us some trouble to add some extra code to switch the flag back and forth, especially when ort-extensions was used as cmake sub-directory for another project.

DCMAKE_MSVC_RUNTIME_LIBRARY is documented feature from cmake and it's kind of a standard that we should follow.

**Review by sayanshaw24 — APPROVED** [review]

File: .pipelines/ci.yml (hunk 1 above)
Comment (sayanshaw24) [reply .pipelines/ci.yml:280]: we could perhaps add a magic string to replace "MultiThreaded$<$<CONFIG:Debug>:Debug>" so it is simple for the user when passing the parameter for DCMAKE_MSVC_RUNTIME_LIBRARY to `build.bat` - and then we can parse it in the YAML file.

**Comment by wenbingl** [comment]
Merge this fixing firstly, and any further improvement are welcomed.

**Pull request merged**